src/Core/Managed/Shared/Extensibility/Implementation/IClock.cs

```diff
@@ -4,6 +4,6 @@
 
     internal interface IClock
     {
-        DateTimeOffset Time { get; }
+        TimeSpan Time { get; }
```

Comment:
was this interface used anywhere else? Why are you returning timespan instead of a number of ticks representing the point in time. First, the name of the property is misleading to what it returns. Second - you are loosing in precision as you converts to the lower-precision value twice. Third - you actually doing two conversions (arithmetic operations) instead of one that is required.